Intermittent test failures can occur when response data spans execute_result responses

From time to time we experience test failures that otherwise work.  The issue occurs when say `stream` responses are returned for a single request due to the way the final response is composed using `'\n'.join()` when `''.join()` is appropriate.  This causes the regular expression match code to assert due to an embedded newline.

Here's an example of the issue:
```
======================================================================
FAIL: test_hello_world (enterprise_gateway.itests.test_scala_kernel.TestScalaKernelLocal)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/kevin-bates/enterprise_gateway/enterprise_gateway/itests/test_scala_kernel.py", line 13, in test_hello_world
    self.assertRegexpMatches(result, 'Hello World')
AssertionError: Regex didn't match: 'Hello World' not found in 'Hel\nlo World\n'

Pulled response from queue for kernel with msg_id: 3596c2f3741f48b49f27637002e4bbc1
{'buffers': [],
 'channel': 'iopub',
 'content': {'name': 'stdout', 'text': 'Hel'},
 'header': {'date': '2018-08-25T03:28:48.696524Z',
            'msg_id': '235a1ed5-5906-4a09-928e-9921a16ca6ab',
            'msg_type': 'stream',
            'session': 'b777e71c-f760-459a-a18e-72857bdb76cc',
            'username': 'elyra',
            'version': '5.0'},
 'metadata': {'timestamp': '1535167728691'},
 'msg_id': '235a1ed5-5906-4a09-928e-9921a16ca6ab',
 'msg_type': 'stream',
 'parent_header': {'msg_id': '3596c2f3741f48b49f27637002e4bbc1',
                   'msg_type': 'execute_request',
                   'session': '',
                   'username': '',
                   'version': '5.0'}}
<<<<<<<<<<<<<<<
Pulled response from queue for kernel with msg_id: 3596c2f3741f48b49f27637002e4bbc1
{'buffers': [],
 'channel': 'iopub',
 'content': {'name': 'stdout', 'text': 'lo World\n'},
 'header': {'date': '2018-08-25T03:28:48.706633Z',
            'msg_id': '8251396e-dfcb-4a48-95e4-08eb6911c37f',
            'msg_type': 'stream',
            'session': 'b777e71c-f760-459a-a18e-72857bdb76cc',
            'username': 'elyra',
            'version': '5.0'},
 'metadata': {'timestamp': '1535167728702'},
 'msg_id': '8251396e-dfcb-4a48-95e4-08eb6911c37f',
 'msg_type': 'stream',
 'parent_header': {'msg_id': '3596c2f3741f48b49f27637002e4bbc1',
                   'msg_type': 'execute_request',
                   'session': '',
                   'username': '',
                   'version': '5.0'}}
```

Updating [this statement](https://github.com/jupyter-incubator/enterprise_gateway/blob/master/enterprise_gateway/client/gateway_client.py#L154) to join on the empty string should do the trick.

--- a/enterprise_gateway/client/gateway_client.py
+++ b/enterprise_gateway/client/gateway_client.py
@@ -151,7 +151,7 @@ class Kernel:
         except BaseException as b:
             print(b)
 
-        return '\n'.join(response)
+        return ''.join(response)
 
     def interrupt(self):
         url = "{}/{}".format(self.kernel_http_api_endpoint, "interrupt")

